cmd/entrypoint/cmd/root.go

```diff
@@ -0,0 +1,202 @@
+	extendDuration    = os.Getenv(constants.ExtendDurationEnvName)
+	slackAgentSvcName = os.Getenv(constants.SlackAgentServiceNameEnvName)
+
+	// Directory/File Paths
+	runnerDir = filepath.Join("/runner")
+	workDir   = filepath.Join(runnerDir, "_work")
+
+	extendFlagFile    = filepath.Join(os.TempDir(), "extend")
+	failureFlagFile   = filepath.Join(os.TempDir(), "failure")
+	cancelledFlagFile = filepath.Join(os.TempDir(), "cancelled")
+	successFlagFile   = filepath.Join(os.TempDir(), "success")
+
+	configCommand = filepath.Join(runnerDir, "config.sh")
+	runSVCCommand = filepath.Join(runnerDir, "bin", "runsvc.sh")
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "entrypoint",
+	Short: "GitHub Actions runner Entrypoint",
+	Long:  "GitHub Actions runner Entrypoint",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		return checkEnvs()
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := os.Mkdir(workDir, 0777); err != nil {
+			return err
+		}
+
+		configArgs := []string{
+			"--unattended",
+			"--replace",
+			"--name", podName,
+			"--url", fmt.Sprintf("https://github.com/%s/%s", runnerOrg, runnerRepo),
+			"--token", runnerToken,
+			"--work", workDir,
+		}
+		well.Go(func(ctx context.Context) error {
+			if err := runCommand(ctx, runnerDir, configCommand, configArgs...); err != nil {
+				return err
+			}
+			if err := runCommand(ctx, runnerDir, runSVCCommand); err != nil {
+				return err
+			}
+
+			extend, err := annotatePods(ctx)
+			if err != nil {
+				return err
+			}
+			if err := slackNotify(ctx, extend); err != nil {
+				return err
+			}
+
+			time.Sleep(time.Duration(1<<63 - 1))
+			return nil
+		})
+		well.Stop()
+		return well.Wait()
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func checkEnvs() error {
+	if len(podName) == 0 {
+		return fmt.Errorf("%s must be set", constants.PodNameEnvName)
+	}
+	if len(podNamespace) == 0 {
+		return fmt.Errorf("%s must be set", constants.PodNamespaceEnvName)
+	}
+	if len(runnerToken) == 0 {
+		return fmt.Errorf("%s must be set", constants.RunnerTokenEnvName)
+	}
+	if len(runnerOrg) == 0 {
+		return fmt.Errorf("%s must be set", constants.RunnerOrgEnvName)
+	}
+	if len(runnerRepo) == 0 {
+		return fmt.Errorf("%s must be set", constants.RunnerRepoEnvName)
+	}
+	if len(extendDuration) == 0 {
+		extendDuration = "20m"
+	}
+
+	// SLACK_AGENT_SERVICE_NAME is optional
+
+	return nil
+}
+
+func runCommand(ctx context.Context, workDir, commandStr string, args ...string) error {
+	command := exec.CommandContext(ctx, commandStr, args...)
+	command.Stdout = os.Stdout
+	command.Stderr = os.Stderr
+	command.Dir = workDir
+	command.Env = removeEnv()
+	if err := command.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func removeEnv() []string {
```

Comment:
I think that name is a bit weird.
For example, how about `removedEnv` or `maskedEnv`?

Reply:
Renamed the `removedEnv`.